openlibrary/plugins/upstream/borrow.py

```diff
@@ -216,7 +216,7 @@ def POST(self, key):
 
     def POST_join_waitinglist(self, edition, user):
         waitinglist.join_waitinglist(user.key, edition.key)
-        raise web.redirect(edition.url("/borrow"))
+        raise web.redirect(edition.url())
```

Comment:
The POST is still to the same place, the difference is the resulting redirect. We're sending the user back to the edition page instead of back to the borrow page.